client/actions/assignments/ui.js

```diff
@@ -536,62 +543,109 @@ const canLinkItem = (item) => (
     )
 );
 
-const openSelectTemplateModal = (assignment) => (
-    (dispatch, getState, {templates, session, desks, notify}) => (
-        dispatch(self.lockAssignment(assignment, 'start_working'))
-            .then((lockedAssignment) => {
-                const currentDesk = desks.getCurrentDesk();
-                const defaultTemplateId = get(currentDesk, 'default_content_template') || null;
-
-                return templates.fetchTemplatesByUserDesk(
-                    session.identity._id,
-                    get(currentDesk, '_id') || null,
-                    1,
-                    200,
-                    'create'
-                ).then((data) => {
-                    let defaultTemplate = null;
-                    const publicTemplates = [];
-                    const privateTemplates = [];
-
-                    (get(data, '_items') || []).forEach((template) => {
-                        if (get(template, '_id') === defaultTemplateId) {
-                            defaultTemplate = template;
-                        } else if (get(template, 'is_public') !== false) {
-                            publicTemplates.push(template);
-                        } else {
-                            privateTemplates.push(template);
+const startWorking = (assignment) => (
```

Comment:
It might be better to split this into two separate actions, for easier maintenance and code readability